### ISSUE TITLE
magisk 26.0 and above crashes home screen if zygisk enabled and you remove magisk app and reboot

### ISSUE THREAD

**Issue by s-lanc44** [topjohnwu/Magisk#7674]
Device: Motorola g7 power
Android version: 9
Magisk version name: 65207f96 (26404) (D)
Magisk version code: 26404

As of 26.0, if you have magisk and magisk app installed with zygisk enabled, and then uninstall the magisk app and reboot, it has a problem as soon as you unlock the phone. the home screen never shows up, and eventually i get an error on the screen saying "System UI is not responding". the screen then goes black (a lit black not a black where there is power). i have no modules installed in magisk. the way to get my phone back was to reboot into bootloader and fastboot the stock boot.img or magisk 25.2 patched image (or a version below 25.2).

If magisk and magisk app are installed and zygisk is disabled, and then i uninstall magisk app and reboot, there are no problems (it adds the stub on the home screen to reinstall magisk)

25.2 works. the problem started as soon as it goes to 26.0. this continues to 26.4, and the debug version (https://github.com/topjohnwu/Magisk/commit/b69804bf9535daaecb1d1cf5fe7e3d39f646aa6c)

here is a link to the [logcat](https://github.com/s-lanc44/Magisk/raw/master/26404debug.txt)

**Comment by yujincheng08** [comment]
please also attach `/cache/magisk.log`

**Comment by s-lanc44** [comment]
here is the magisk log
https://github.com/s-lanc44/Magisk/blob/master/magisk.log.bak

i generated a new logcat to accompany the magisk log
https://github.com/s-lanc44/Magisk/blob/master/26404debug2.txt

**Comment by s-lanc44** [comment]
also wanted to note that if i go to the debug version and have zygisk enabled and i remove the magisk app, i dont even need to reboot it to get it to crash. if i remove the magisk app and try to use an app that wants to be rooted, i am subsequently unable to open apps in my home screen (and am forced to reboot and flash an older magisk version)

**Comment by yujincheng08** [comment]
run `adb bugreport` and send the generated file.

**Comment by yujincheng08** [comment]
never mind. know whats going on now.

**Comment by yujincheng08** [comment]
try artifacts from https://github.com/topjohnwu/Magisk/pull/7677

**Comment by s-lanc44** [comment]
I built the apk from source based on your branch, and it works!
I couldn't believe it! I also checked the situation where you remove the app, dont restart, but you open an app that requests superuser. That is working properly now too (it adds the stub if its not there, or it opens the stub app)

Fantastic work Yujin. Thanks for you and your whole team for what you have made. 

Unrelated: building the apk for the first time was tough! I was on a company computer, so i had to deal with all the certificate stuff (git, cargo, gradle). But now i know i can do it!

**Comment by s-lanc44** [comment]
for anyone who stumbles upon this and is looking for a workaround until this is added to the next official build:

you can always press the button in the magisk settings to hide the magisk manager app and rename it. ive gone step further and hid/renamed the app and i used app freezer to freeze the app. your magisk settings will stay as is. you wont be able to use magisk for superuser until you uninstall the app and reinstall.

my use case is i need the magisk manager app uninstalled or frozen, but i also need any magisk changes i had made (editing hosts file or freezing apps) to remain in place.

**Comment by FerhatStl** [comment]
I happened to got this error too. My Mi 9T Pro stuck at home screen. Nothing loaded to top panel and phone was not unlocking. I returned back to Magisk v25.2 to get rid of problem It works fine right now. Also Twrp is not capable of decrypting my data right now. So I can't access phone data within TWRP.

**Comment by s-lanc44** [comment]
> I happened to got this error too. My Mi 9T Pro stuck at home screen. Nothing loaded to top panel and phone was not unlocking. I returned back to Magisk v25.2 to get rid of problem It works fine right now. Also Twrp is not capable of decrypting my data right now. So I can't access phone data within TWRP.

Glad to hear it! Sucks about TWRP hope you get that resolved

**Comment by fizzgnar** [comment]
I also encountered the same issue. Everything was fine in version 26.3, but after updating to version 26.4 and restarting, I experienced a black screen. Interestingly, I could unlock through gesture swipe and pull down the menu (with graphical UI but no text), and after about ten minutes, the text slowly started appearing. Yes, it didn't completely freeze but was loading very slowly.

**Comment by gitdine** [comment]
Similar issue here.
After installing Magisk v26.4 and enabling zygisk on my Poco X3 Pro, Android became completely broken after reboot. Couldn't even get to the home screen because system processes kept crashing. I was able to reproduce this several times while troubleshooting.

So I reinstalled Android, and installed the older Magisk v26.3 and enabled zygisk - this time it worked flawlessly, no crashing.

Seems like Magisk v26.4 has a serious problem with zygisk on certain devices.
